backend/src/API/coffeeChatAPI.ts

```diff
@@ -145,61 +179,135 @@ export const checkMemberMeetsCategory = async (
+      const isTpm = otherMember.role === 'tpm';
+      const notSameTeam = haveNoCommonSubteams(submitter, otherMember);
+      status = isTpm && notSameTeam ? 'pass' : 'fail';
+      if (status === 'fail') {
+        if (!isTpm && !notSameTeam) {
+          message = `${otherMember.firstName} ${otherMember.lastName} is not a TPM and is on the same team as ${submitter?.firstName} ${submitter?.lastName}`;
+        } else if (!isTpm) {
+          message = `${otherMember.firstName} ${otherMember.lastName} is not a TPM`;
+        } else if (!notSameTeam) {
+          message = `${otherMember.firstName} ${otherMember.lastName} is a TPM, but is on the same team as ${submitter?.firstName} ${submitter?.lastName}`;
+        }
+      }
+    }
+
+    // If otherMemberProperties doesn't exist, status should stay undefined
+    if (otherMemberProperties) {
+      if (category === 'a newbie') {
+        status = otherMemberProperties.newbie ? 'pass' : 'fail';
+        status === 'fail' &&
+          (message = `${otherMember.firstName} ${otherMember.lastName} is not a newbie`);
+      } else if (category === 'is/was a TA') {
+        status = otherMemberProperties.ta ? 'pass' : 'fail';
+        status === 'fail' &&
+          (message = `${otherMember.firstName} ${otherMember.lastName} was never a TA`);
+      } else if (category === 'major/minor that is not cs/infosci') {
+        status = otherMemberProperties.notCsOrInfosci ? 'pass' : 'fail';
+        status === 'fail' &&
+          (message = `${otherMember.firstName} ${otherMember.lastName} is a CS or Infosci major`);
+      }
+
+      // If submitterProperties doesn't exist, status should stay undefined
+      if (submitterProperties) {
+        if (category === 'from a different college') {
+          status = otherMemberProperties.college !== submitterProperties.college ? 'pass' : 'fail';
+          status === 'fail' &&
+            (message = `${otherMember.firstName} ${otherMember.lastName} is from the same college as ${submitter.firstName} ${submitter.lastName} (${otherMemberProperties.college})`);
+        }
+      }
+
+      if (category === 'a lead (not your role)') {
+        const isLead = otherMember.role === 'lead';
+        let diffRole;
+
+        if (submitter.role !== 'lead') {
+          diffRole = otherMemberProperties.leadType !== submitter.role;
+        } else if (submitterProperties) {
+          diffRole = otherMemberProperties.leadType !== submitterProperties.leadType;
+        } else {
+          diffRole = undefined;
+        }
+
+        // If submitter is a lead, but submitterProperties doesn't exist, status should stay undefined
+        if (diffRole !== undefined) {
+          status = isLead && diffRole;
+
+          if (!status) {
+            if (!isLead && !diffRole) {
+              message = `${otherMember.firstName} ${otherMember.lastName} is not a lead and is from the same role as ${submitter.firstName} ${submitter.lastName}`;
```

Comment:
Not being a lead should be enough to disqualify the coffee chat. I probably wouldn't add the part about being in the same role.